### PR TITLE
Parameter validation for GCM

### PR DESCRIPTION
**Pull request by k-stachowiak**
Based on #1643, this PR adds parameter validation for the GCM module.

### CODE REVIEW

File: include/mbedtls/gcm.h [hunk 1]
```diff
@@ -136,21 +138,28 @@ int mbedtls_gcm_setkey( mbedtls_gcm_context *ctx,
  *                    calling this function in decryption mode.
  * \param length    The length of the input data, which is equal to the length
  *                  of the output data.
- * \param iv        The initialization vector.
+ * \param iv        The initialization vector. This must be a readable buffer of
+ *                  at least \p iv_len Bytes.
  * \param iv_len    The length of the IV.
- * \param add       The buffer holding the additional data.
+ * \param add       The buffer holding the additional data. If \p add_len is
```
Comment (hanno-arm) [inline include/mbedtls/gcm.h]: Minor: I'd prefer to clarify that this may be `\c NULL` if `\p add_len` is `\c 0`.

File: include/mbedtls/gcm.h [hunk 2]
```diff
@@ -136,21 +138,28 @@ int mbedtls_gcm_setkey( mbedtls_gcm_context *ctx,
...
+ *                  greater than zero, this must be of at least that size in
+ *                  Bytes.
  * \param add_len   The length of the additional data.
- * \param input     The buffer holding the input data. Its size is \b length.
- * \param output    The buffer for holding the output data. It must have room
- *                  for \b length bytes.
+ * \param input     The buffer holding the input data. If \p length is greater
```
Comment (hanno-arm) [inline include/mbedtls/gcm.h]: Same here.

**Comment by k-stachowiak** [comment]
@hanno-arm I have rebased the PR. Could you please re-reveiw it?

**Comment by mpg** [comment]
CI only failed in known-flaky timing test, so as good as a pass.

**Comment by k-stachowiak** [comment]
@hanno-arm @mpg The PR has been rebased against development.

**Comment by sbutcher-arm** [comment]
PR #1643 has been merged, so removing the 'needs:preceding PR' label.

Reviewed, approved and passes CI, so marking as 'ready for merge'.

**Pull request merged**